src/Sarif/UriHelper.cs

```diff
@@ -78,5 +90,36 @@ public static string MakeValidUri(string path)
 
             return validUri;
         }
+
+        private static readonly Regex s_oneDotPattern =
+            new Regex(@"^\.[\\/]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        private static readonly Regex s_twoDotPattern =
+            new Regex(@"^\.\.[\\/]", RegexOptions.Compiled | RegexOptions.CultureInvariant);
+
+        private static string GetLeadingDotSegments(string path)
+        {
+            var sb = new StringBuilder();
+
+            bool moreDotSegments = true;
+            while (moreDotSegments)
+            {
+                if (s_oneDotPattern.IsMatch(path))
+                {
+                    path = path.Substring(2);
+                }
+                else if (s_twoDotPattern.IsMatch(path))
+                {
+                    path = path.Substring(3);
+                    sb.Append("../");
+                }
+                else
+                {
+                    moreDotSegments = false;
```

Comment:
I think we miss the scenario where the input path is just a dot segment (".."). 
... However, I don't think there's a valid SARIF scenario for this because ".." would indicate a path, and I think all the SARIF URIs are files?